apps/jobs/views.py

```diff
@@ -2301,26 +2302,152 @@ def get_submission_file_presigned_url(request, challenge_phase_pk):
+            "upload_id": response.get("upload_id"),
             "submission_pk": submission.pk,
         }
         return Response(response_data, status=status.HTTP_201_CREATED)
     response_data = {"error": serializer.errors}
     return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
 
+@api_view(["POST"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def submission_file_upload_complete(request, challenge_phase_pk, submission_pk):
+    """
+    API to complete multipart upload of presigned url submission
+
+    Arguments:
+        request {HttpRequest} -- The request object
+        challenge_phase_pk {int} -- Challenge phase primary key
+        submission_pk {int} -- Submission primary key
+    Returns:
+         Response Object -- An object containing the presignd url and submission id, or an error message if some failure occurs
+    """
+    if settings.DEBUG or settings.TEST:
+        response_data = {"error": "Sorry, this feature is not available in development or test environment."}
+        return Response(response_data)
+
+    # Check if the challenge phase exists or not
+    try:
+        challenge_phase = get_challenge_phase_model(challenge_phase_pk)
+    except ChallengePhase.DoesNotExist:
+        response_data = {"error": "Challenge Phase does not exist"}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    challenge = challenge_phase.challenge
+
+    participant_team_id = get_participant_team_id_of_user_for_a_challenge(
+        request.user, challenge.pk
+    )
+
+    if not challenge.is_active:
+        response_data = {"error": "Challenge is not active"}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check if challenge phase is active
+    if not challenge_phase.is_active:
+        response_data = {
+            "error": "Sorry, cannot accept submissions since challenge phase is not active"
+        }
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check if user is a challenge host or a participant
+    if not is_user_a_host_of_challenge(request.user, challenge.pk):
+        # Check if challenge phase is public and accepting solutions
+        if not challenge_phase.is_public:
+            response_data = {
+                "error": "Sorry, cannot accept submissions since challenge phase is not public"
+            }
+            return Response(response_data, status=status.HTTP_403_FORBIDDEN)
+
+        if not challenge.approved_by_admin:
+            response_data = {
+                "error": "Challenge is not yet approved by admin."
+            }
+            return Response(
+                response_data, status=status.HTTP_406_NOT_ACCEPTABLE
+            )
+
+        # if allowed email ids list exist, check if the user exist in that list or not
```

Comment:
We also have a blocked email Ids list and we should also check for that along with checking for allowed and blocked domains.

Reply:
Check here https://github.com/Cloud-CV/EvalAI/pull/3052/files#diff-f56c8d0075d862c83b8c37b0e742aec33542a7f1e9681b2cebdfc6306e8ee920R2411 it is already added